### PR TITLE
Remove flags, as they are not needed according to Bluetooth 4.2 specification

### PR DESCRIPTION
**Pull request by elwerene**
Bluetooth specification 4.2:

```
9.2.2.2 Conditions
A device in the non-discoverable mode that sends advertising events shall not
set the ‘LE General Discoverable Mode’ flag or ‘LE Limited Discoverable Mode’
flag in the Flags AD type (see [Core Specification Supplement], Part A, Section
1.3). A Peripheral device in the non-connectable mode may send non-
connectable undirected advertising events or scannable undirected advertising
events or may not send advertising packets.
```

Sounds to me like sending no Flags AD is totally fine and correct (and saves 3 bytes).

### CODE REVIEW

**Comment by jonas-schievink** [comment]
That doesn't say that it is allowed to leave out the entire flags byte though. The flags structure also has a `BR_EDR_NOT_SUPPORTED` flag which we set. If that flag is required when classic BT is not supported, then we cannot leave out the flags.

**Comment by jonas-schievink** [comment]
Ah nevermind, the Core Specification Supplement says:

> The Flags data type shall be included when any of the Flag bits are non-zero and the advertising packet is connectable, otherwise the Flags data type may be omitted.

File: rubble/src/link/advertising.rs [hunk 1]
```diff
@@ -570,17 +570,15 @@ impl PduBuf {
...
     pub fn beacon(
         advertiser_addr: DeviceAddress,
         advertiser_data: &[AdStructure<'_>],
     ) -> Result<Self, Error> {
         Self::adv(
             PduType::AdvNonconnInd,
             advertiser_addr,
-            &mut iter::once(&AdStructure::from(Flags::broadcast())).chain(advertiser_data),
+            &mut advertiser_data.iter(),
         )
```
Comment (jonas-schievink) [inline rubble/src/link/advertising.rs:582]: The body can just forward to `nonconnectable_undirected` now

Comment (elwerene) [reply rubble/src/link/advertising.rs:582]: I'll change it

Comment (elwerene) [reply rubble/src/link/advertising.rs]: done

**Pull request merged**